pkg/controller/vsphere/reconciler.go

```diff
@@ -616,7 +617,13 @@ func getDiskSpec(s *machineScope, devices object.VirtualDeviceList) (types.BaseV
 	}
 
 	disk := disks[0].(*types.VirtualDisk)
-	disk.CapacityInKB = int64(s.providerSpec.DiskGiB) * 1024 * 1024
+	cloneCapacityKB := int64(s.providerSpec.DiskGiB) * 1024 * 1024
+	if disk.CapacityInKB > cloneCapacityKB {
```

Comment:
Should we not be requiring them to be the same? Does it being smaller not also cause an issue?

Reply:
No, expanding the disk is a supported operation.

Reply:
I was looking at this backwards, ignore me